security/pkg/nodeagent/sds/adapter.go

```diff
@@ -0,0 +1,54 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sds
+
+import (
+	"context"
+
+	xdsapiv2 "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	discoveryv2 "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v2"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	v2 "istio.io/istio/pilot/pkg/proxy/envoy/v2"
+)
+
+// sdsServiceAdapter is a sdsservice that converts v3 Discovery messages to v2 messages.
+// See notes in DiscoveryStreamV2Adapter for more info.
+type sdsServiceAdapter struct {
+	s *sdsservice
+}
+
+func (d *sdsServiceAdapter) FetchSecrets(ctx context.Context, request *xdsapiv2.DiscoveryRequest) (*xdsapiv2.DiscoveryResponse, error) {
+	v3Resp, err := d.s.FetchSecrets(ctx, v2.UpgradeV2Request(request))
+	return v2.DowngradeV3Response(v3Resp), err
+}
+
+// We implement the v2 ADS API
+var _ discoveryv2.SecretDiscoveryServiceServer = &sdsServiceAdapter{}
```

Comment:
What is the purpose of this step?

Reply:
support backwards compatibly with old pilot